JitsiConference.js

```diff
@@ -1623,7 +1786,519 @@ JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
+    if (session.isP2P === this.p2pEstablished) {
+        this.eventEmitter.emit(JitsiConferenceEvents.CONNECTION_INTERRUPTED);
+    }
+};
+
+/**
+ * Handles {@link XMPPEvents.CONNECTION_ICE_FAILED}
+ * @param {JingleSessionPC} session
+ * @private
+ */
+JitsiConference.prototype._onIceConnectionFailed = function(session) {
+    if (session.isP2P) {
+        this._stopP2PSession('connectivity-error', 'ICE FAILED');
+    }
+};
+
+/**
+ * Handles {@link XMPPEvents.CONNECTION_RESTORED}
+ * @param {JingleSessionPC} session
+ * @private
+ */
+JitsiConference.prototype._onIceConnectionRestored = function(session) {
+    if (session.isP2P) {
+        this.isP2PConnectionInterrupted = false;
+    } else {
+        this.isJvbConnectionIsInterrupted = false;
+    }
+
+    if (session.isP2P === this.p2pEstablished) {
+        this.eventEmitter.emit(JitsiConferenceEvents.CONNECTION_RESTORED);
+    }
+};
+
+/**
+ * Accept incoming P2P Jingle call.
+ * @param {JingleSessionPC} jingleSession the session instance
+ * @param {jQuery} jingleOffer a jQuery selector pointing to 'jingle' IQ element
+ * @private
+ */
+JitsiConference.prototype._acceptP2PIncomingCall
+= function(jingleSession, jingleOffer) {
+    jingleSession.setSSRCOwnerJid(this.room.myroomjid);
+
+    this.isP2PConnectionInterrupted = false;
+
+    // Accept the offer
+    this.p2pJingleSession = jingleSession;
+
+    this.p2pJingleSession.initialize(
+        false /* initiator */, this.room, this.rtc);
+
+    const localTracks = this.getLocalTracks();
+
+    logger.debug(`Adding ${localTracks} to P2P...`);
+    this.p2pJingleSession.addLocalTracks(localTracks).then(
+        () => {
+            logger.debug(`Add ${localTracks} to P2P done!`);
+            this.p2pJingleSession.acceptOffer(
+                jingleOffer,
+                () => {
+                    logger.debug('Got RESULT for P2P "session-accept"');
+                },
+                error => {
+                    logger.error(
+                        'Failed to accept incoming P2P Jingle session', error);
+                }
+            );
+        },
+        error => {
+            logger.error(
+                `Failed to add ${localTracks} to the P2P connection`, error);
+        });
+};
+
+/**
+ * Attaches local tracks back to the JVB connection.
+ * @private
+ */
+JitsiConference.prototype._attachLocalTracksToJvbSession = function() {
+    const localTracks = this.getLocalTracks();
+
+    logger.info(`Attaching ${localTracks} to JVB`);
+    this.jvbJingleSession.attachLocalTracks(localTracks).then(
+        () => {
+            logger.info(`Attach ${localTracks} to JVB success!`);
+        },
+        error => {
+            logger.error(`Attach ${localTracks} to JVB failed!`, error);
+        });
+};
+
+/**
+ * Adds remote tracks to the conference associated with the JVB session.
+ * @private
+ */
+JitsiConference.prototype._addRemoteJVBTracks = function() {
+    this._addRemoteTracks('JVB', this.jvbJingleSession);
```

Comment:
there is a check if "this.jvbJingleSession" is defined which print a log in case it's not:
https://github.com/jitsi/lib-jitsi-meet/blob/master/JitsiConference.js#L1950

Anyway I don't really understand what's wrong with passing JingleSessionPC instance there and how passing remote tracks makes it any better.